internal/weapons/claymore/forestregalia/forestregalia.go

```diff
@@ -0,0 +1,93 @@
+package forestregalia
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/core/player/weapon"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+func init() {
+	core.RegisterWeaponFunc(keys.ForestRegalia, NewWeapon)
+}
+
+type Weapon struct {
+	Index int
+}
+
+func (w *Weapon) SetIndex(idx int) { w.Index = idx }
+func (w *Weapon) Init() error      { return nil }
+
+const (
+	icdKey  = "forestregalia-icd"
+	buffKey = "forest-sanctuary"
```

Comment:
Pls rename the buffKey variables to the weapon names to keep in line with most of the other weapon buffs. Also, having the Forest Regalia and Sapwood Blade buff names overlap isn't good because that means these buffs overwrite each other instead of stacking if a character happens to pick up a leaf from both Forest Regalia and Sapwood Blade.

Reply:
Nvm, I just noticed that the description says these buffs can't stack...